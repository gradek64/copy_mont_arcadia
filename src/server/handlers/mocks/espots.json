{
  "pageId": "116711",
  "pageName": "Mobile - eSpot Page Mapping",
  "baseline": "3",
  "pageLastPublished": "2016-05-31 16:13:36.06833",
  "pagePublishedBy": "Gary King",
  "contentPath": "/cms/pages/json/json-0000116711/json-0000116711.json",
  "seoUrl": "",
  "templateName": "Monty eSpot Map (MASTER) (REF 0000012483)",
  "templatePublishDate": "2016-05-16 12:15:49.613042",
  "version": "1",
  "pageMap": [
    null, {
      "pageId": "116870",
      "pageName": "mobilePLPESpotPos1",
      "baseline": "3",
      "lastPublished": "2016-05-31 16:14:12.029926",
      "seoUrl": "",
      "contentPath": "cms/pages/json/json-0000116870/json-0000116870.json"
    }, {
      "pageId": "116710",
      "pageName": "mobilePDPESpotPos1",
      "baseline": "4",
      "lastPublished": "2016-06-01 17:02:11.479811",
      "seoUrl": "",
      "contentPath": "cms/pages/json/json-0000116710/json-0000116710.json"
    }, {
      "pageId": "116709",
      "pageName": "mobilePDPESpotPos2",
      "baseline": "4",
      "lastPublished": "2016-06-01 17:01:59.996714",
      "seoUrl": "",
      "contentPath": "cms/pages/json/json-0000116709/json-0000116709.json"
    }, {
      "pageId": "116871",
      "pageName": "mobileShoppingBagESpotPos1",
      "baseline": "3",
      "lastPublished": "2016-05-31 16:14:16.048327",
      "seoUrl": "",
      "contentPath": "cms/pages/json/json-0000116871/json-0000116871.json"
    }, {
      "pageId": "116874",
      "pageName": "mobileCheckoutESpotPos1",
      "baseline": "3",
      "lastPublished": "2016-05-31 16:14:21.114911",
      "seoUrl": "",
      "contentPath": "cms/pages/json/json-0000116874/json-0000116874.json"
    }, {
      "pageId": "116875",
      "pageName": "mobileOrderConfirmationESpotPos1",
      "baseline": "3",
      "lastPublished": "2016-05-31 16:14:24.237599",
      "seoUrl": "",
      "contentPath": "cms/pages/json/json-0000116875/json-0000116875.json"
    }, {
      "pageId": "116708",
      "pageName": "mobileTacticalMessageESpotPos1",
      "baseline": "4",
      "lastPublished": "2016-06-01 17:01:53.606235",
      "seoUrl": "",
      "contentPath": "cms/pages/json/json-0000116708/json-0000116708.json"
    }
  ]
}
