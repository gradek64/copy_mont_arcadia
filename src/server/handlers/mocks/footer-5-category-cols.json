{
  "newsletter": {
    "label": "Sign up and get 20% off",
    "placeholder": "Enter your email address",
    "button": "Sign Up",
    "signUpUrl": "/en/tsuk/category/help-information-5634539/sign-up-to-style-notes-5651318",
    "openNewWindow": false
  },
  "categories": [
    {
      "label": "HELP",
      "links": [
        {
          "label": "Store Finder",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "FAQs",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Contact us",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Size Guide",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Washcare & Aftercare",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Accessibility",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Site Map",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Topshop at Oxford Circus",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        }
      ]
    },
    {
      "label": "DELIVERY & RETURNS",
      "links": [
        {
          "label": "Track your order",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Delivery",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Returns",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "E-receipts",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        }
      ]
    },
    {
      "label": "CARDS & OFFERS",
      "links": [
        {
          "label": "Student Discount",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Gift Cards",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Topshop Card",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        }
      ]
    },
    {
      "label": "ABOUT US",
      "links": [
        {
          "label": "Topshop Supports",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Fashion Footprint",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Careers & Opportunities",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "T&Cs",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Modern Slavery Act",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Topshop at London Fashion Week",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        }
      ]
    },
    {
      "label": "ABOUT US",
      "links": [
        {
          "label": "Topshop Supports",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Fashion Footprint",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Careers & Opportunities",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "T&Cs",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Modern Slavery Act",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        },
        {
          "label": "Topshop at London Fashion Week",
          "linkUrl": "/en/tsuk/category/store-finder-1824332/home?cat2=1102568&intcmpid=W_FOOTER_MAIN_%20STORE_FINDER",
          "openNewWindow": true
        }
      ]
    }
  ]
}
