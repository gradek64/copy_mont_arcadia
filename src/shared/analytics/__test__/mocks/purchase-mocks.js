export const basicState = {
  account: {
    user: {
      email: 'aamir.ahmad@arcadiagroup.co.uk',
    },
  },
  config: {
    currencyCode: 'EUR',
  },
  checkout: {
    orderCompleted: {
      userId: '987654',
      orderId: 1706986,
      deliveryMethod: 'UK Standard up to 4 working days',
      deliveryCost: '4.00',
      deliveryPrice: '4.00',
      totalOrderPrice: '44.8',
      totalOrdersDiscount: '-£7.2',
      deliveryAddress: {
        address1: '2 Carlile Close',
        name: 'Mr 3D. AUTHORISED',
        country: 'United Kingdom',
        address2: 'LONDON',
        address3: 'E3 2ES',
      },
      ddpPromotion: {
        label: 'DDP_Shipping_Promotion',
        value: -5.95,
      },
      orderLines: [
        {
          skuId: '123456789',
          lineNo: '26M48MIVR',
          colour: 'IVORY',
          size: '6',
          name: 'PETITE Floral Jacquard Midi Wrap Dress',
          quantity: 1,
          unitPrice: '30.00',
          brand: 'Topshop',
          category: 'Dresses',
          reviewRating: '4',
          ecmcCategory: 'DRES',
          department: '21',
        },
        {
          skuId: '987654321',
          lineNo: '42H08KGLD',
          colour: 'GOLD',
          size: '38',
          name: 'HELD UP Leather Knot Sandals',
          quantity: 1,
          unitPrice: '18.00',
          brand: 'Topshop',
          category: 'Shoes',
          reviewRating: '1',
          ecmcCategory: 'SHOE',
          department: '24',
        },
      ],
      paymentDetails: [
        {
          paymentMethod: 'Visa',
        },
      ],
      promoCodes: [],
      productRevenue: '73.00',
    },
    orderError: false,
  },
}

export const ssrMockState = {
  account: {
    user: {
      email: 'aamir.ahmad@arcadiagroup.co.uk',
    },
  },
  config: {
    currencyCode: 'EUR',
  },
  checkout: {
    orderCompleted: {
      userId: '987654',
      orderId: 2337940,
      subTotal: '130.00',
      returnPossible: false,
      returnRequested: false,
      deliveryMethod: 'Friday Nominated Day Delivery:',
      deliveryDate: 'Friday 23 March 2018',
      deliveryCost: '6.00',
      deliveryCarrier: 'Home Delivery Network',
      deliveryPrice: '6.00',
      totalOrderPrice: '136.00',
      totalOrdersDiscountLabel: '',
      totalOrdersDiscount: '',
      billingAddress: {
        address1: '2 Carlile Close',
        name: 'Mr 3D. AUTHORISED',
        country: 'United Kingdom',
        address2: 'LONDON',
        address3: 'E3 2ES',
      },
      deliveryAddress: {
        address1: '2 Carlile Close',
        name: 'Mr 3D. AUTHORISED',
        country: 'United Kingdom',
        address2: 'LONDON',
        address3: 'E3 2ES',
      },
      orderLines: [
        {
          skuId: '29689232',
          lineNo: '02K77LMDT',
          name: 'MOTO Mid Blue Lace Fly Jamie Jeans',
          size: 'W2630',
          colour: 'MID STONE',
          imageUrl:
            'https://ts.stage.arcadiagroup.ltd.uk/wcsstore/TopShop/images/catalog/TS02K77LMDT_Small_M_1.jpg',
          quantity: 2,
          unitPrice: '42.00',
          discount: '',
          discountPrice: '0.00',
          total: '84.00',
          nonRefundable: false,
          brand: 'Topshop',
          category: 'Trousers',
          reviewRating: '3',
          ecmcCategory: 'TROU',
          department: '13',
        },
        {
          skuId: '29832681',
          lineNo: '20B03MPPK',
          name: 'OhK! Mini Tissues',
          size: '',
          colour: 'PALE PINK',
          imageUrl:
            'https://ts.stage.arcadiagroup.ltd.uk/wcsstore/TopShop/images/catalog/TS20B03MPPK_Small_M_1.jpg',
          quantity: 4,
          unitPrice: '1.00',
          discount: '',
          discountPrice: '0.00',
          total: '4.00',
          nonRefundable: false,
          brand: 'Topshop',
          category: 'Miscellaneous',
          reviewRating: '5',
          ecmcCategory: 'MISC',
          department: '12',
        },
        {
          skuId: '31305407',
          lineNo: '02K06NBLG',
          name: 'MOTO Jamie Skinny Fit Jeans',
          size: 'W3234',
          colour: 'Blue',
          imageUrl:
            'https://ts.stage.arcadiagroup.ltd.uk/wcsstore/TopShop/images/catalog/TS02K06NBLG_Small_M_1.jpg',
          quantity: 1,
          unitPrice: '42.00',
          discount: '',
          discountPrice: '0.00',
          total: '42.00',
          nonRefundable: false,
          brand: 'Topshop',
          category: 'Trousers',
          reviewRating: '5',
          ecmcCategory: 'TROU',
          department: '34',
        },
      ],
      paymentDetails: [
        {
          paymentMethod: 'Visa',
          cardNumberStar: '************1111',
          totalCost: '£136.00',
        },
      ],
      currencyConversion: {
        currencyRate: 'GBP',
      },
      returning_buyer: false,
      eMarketingEspot1URL: {
        contentText:
          '%3C%21--+New+Customer+Trigger--%3E%0A%3Cscript+type%3D%22text%2Fjavascript%22%3E%0Avar+existingcust+%3D+%22Sale-New%22%0A%3C%2Fscript%3E',
      },
      promoCodes: [],
      productRevenue: '130.00',
      ddpPromotion: {
        ddpPromotion: 'DDP_Shipping_Promotion',
        value: -5.95,
      },
    },
    orderSummary: {},
    orderSummaryError: {},
    orderError: false,
    isDeliverySameAsBilling: true,
    useDeliveryAsBilling: true,
    verifyPayment: false,
    showExpressDeliveryOptionsOnSummary: false,
    editingDetails: false,
    storeUpdating: false,
    deliveryAndPayment: {
      deliveryEditingEnabled: false,
    },
    savePaymentDetails: false,
  },
}
