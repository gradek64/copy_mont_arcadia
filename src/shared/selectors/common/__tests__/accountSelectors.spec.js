import {
  getUser,
  getUserDetails,
  getUserAddress,
  getUserCountry,
  getUserEmail,
  getHashedUserEmail,
  selectStoredPaymentDetails,
  getMyCheckoutDetailsEditingEnabled,
  getMyCheckoutDetailForm,
  getMCDPaymentMethodForm,
  isSaveMyCheckoutDetailsDisabled,
  selectDeliveryCountry,
  selectBillingCountry,
  getPaymentCardDetailsMCD,
  getDeliveryDetails,
  getOrderDetailsDeliveryPostCode,
  getValidUKOrdersCountries,
  getValidOrderStatusReturn,
  isResetPasswordLinkValid,
  getCreditCardType,
  isKlarnaDefaultPaymentType,
  getUserTrackingId,
  isLoggedIn,
  getExponeaMemberId,
} from '../accountSelectors'
import myAccountMock from '../../../../../test/mocks/myAccount-response.json'
import { paymentMethodsList } from '../../../../../test/mocks/paymentMethodsMocks'
import myCheckoutDetailsMocks, {
  myCheckoutDetailsErrors,
  myCheckoutDetailsNoErrors,
  myCheckoutDetailsCVVError,
} from '../../../../../test/mocks/forms/myCheckoutDetailsFormsMocks'
import configMock from '../../../../../test/mocks/config'
import deepFreeze from 'deep-freeze'

const snapshot = (action) => expect(action).toMatchSnapshot()

describe('@Account Selectors', () => {
  const state = {
    config: configMock,
    account: {
      user: myAccountMock,
      myCheckoutDetails: {
        editingEnabled: false,
      },
      resetPasswordLinkIsValid: false,
    },
    paymentMethods: paymentMethodsList,
    forms: {
      account: {
        myCheckoutDetails: myCheckoutDetailsMocks,
      },
      // @todo: move that behind forms.acount.myCheckoutDetails
      checkout: {
        billingCardDetails: {
          fields: {
            paymentType: {
              value: 'VISA',
              isDirty: false,
              isTouched: false,
              isFocused: false,
            },
          },
        },
      },
    },
    findAddress: {
      isManual: false,
    },
  }

  describe('[getUser selector]', () => {
    it('return empty when state is empty', () => {
      expect(getUser({})).toEqual({})
    })
    it('return user from account', () => {
      expect(getUser(state)).toEqual(myAccountMock)
    })
  })

  describe(getUserEmail, () => {
    it('returns the users email address if available', () => {
      const state = {
        account: {
          user: {
            email: 'aamir.ahmad@arcadiagroup.co.uk',
          },
        },
      }
      expect(getUserEmail(state)).toEqual('aamir.ahmad@arcadiagroup.co.uk')
    })
    it('returns null if there is no user email address', () => {
      const state = {}
      expect(getUserEmail(state)).toEqual(null)
    })
  })

  describe(getHashedUserEmail, () => {
    it('returns the users hashed email address if available', () => {
      const state = {
        account: {
          user: {
            email: 'aamir.ahmad@arcadiagroup.co.uk',
          },
        },
      }
      expect(getHashedUserEmail(state)).toEqual(
        '0904c4fabd5ce6902f0484bf47801851814336a7f6a6f691857919554b80ecf2'
      )
    })
    it('returns null if there is no user email address', () => {
      const state = {}
      expect(getHashedUserEmail(state)).toEqual(null)
    })
  })

  describe('getUserTrackingId selector', () => {
    it('should return undefined if no user defined', () => {
      expect(getUserTrackingId({})).toBeUndefined()
    })
    it('should return userTrackingId if available on user', () => {
      expect(
        getUserTrackingId({
          account: {
            user: {
              userTrackingId: 123456,
            },
          },
        })
      ).toBe(123456)
    })
  })

  describe('[isResetPasswordLinkValid selector', () => {
    it('return isResetPasswordLinkValid value', () => {
      expect(isResetPasswordLinkValid(state)).toMatchSnapshot()
    })
    it('returns isResetPasswordLinkValid value as false by default', () => {
      expect(isResetPasswordLinkValid(state)).toEqual(false)
    })
    it('returns isResetPasswordLinkValid value as true if set as true', () => {
      const state = {
        account: {
          resetPasswordLinkIsValid: true,
        },
      }
      expect(isResetPasswordLinkValid(state)).toEqual(true)
    })
  })

  describe('getDeliveryDetails', () => {
    it('returns an empty object by default', () => {
      expect(getDeliveryDetails({})).toEqual({})
    })

    it('returns the delivery details', () => {
      expect(getDeliveryDetails(state)).toMatchSnapshot()
    })
  })

  describe('[getUserDetails selector]', () => {
    it('return empty when state is empty', () => {
      expect(getUserDetails({}, 'deliveryDetails')).toEqual({})
    })

    it('return empty when type is empty', () => {
      expect(getUserDetails({}, '')).toEqual({})
      expect(getUserDetails({}, undefined)).toEqual({})
      expect(getUserDetails({}, null)).toEqual({})
      expect(getUserDetails(state, '')).toEqual({})
      expect(getUserDetails(state, undefined)).toEqual({})
      expect(getUserDetails(state, null)).toEqual({})
    })

    it('extract userDetails from user deliveryDetails', () => {
      snapshot(getUserDetails(state, 'deliveryDetails'))
    })

    it('extract userDetails from user billingDetails', () => {
      snapshot(getUserDetails(state, 'billingDetails'))
    })
  })

  it('[getUserAddress selector] return empty when state is empty', () => {
    expect(getUserAddress({}, 'deliveryDetails')).toEqual({})
  })

  it('[getUserAddress selector] return empty when type is empty', () => {
    expect(getUserAddress({}, '')).toEqual({})
    expect(getUserAddress({}, undefined)).toEqual({})
    expect(getUserAddress({}, null)).toEqual({})
    expect(getUserAddress(state, '')).toEqual({})
    expect(getUserAddress(state, undefined)).toEqual({})
    expect(getUserAddress(state, null)).toEqual({})
  })

  it('[getUserAddress selector] extract user Address from deliveryDetails', () => {
    snapshot(getUserAddress(state, 'deliveryDetails'))
  })

  it('[getUserAddress selector] extract user Address from billingDetails', () => {
    snapshot(getUserAddress(state, 'billingDetails'))
  })

  it('[get User Countries], extract delivery country', () => {
    snapshot(getUserCountry(state, 'deliveryDetails'))
  })

  it('[get User Countries], extract billing country', () => {
    snapshot(getUserCountry(state, 'billingDetails'))
  })

  it('[get User Countries selector] return empty when state is empty', () => {
    expect(getUserCountry({}, 'deliveryDetails')).toEqual('United Kingdom')
  })

  it('[get User Countries], return empty when type is empty', () => {
    expect(getUserCountry({}, '')).toEqual('United Kingdom')
    expect(getUserCountry({}, undefined)).toEqual('United Kingdom')
    expect(getUserCountry({}, null)).toEqual('United Kingdom')
    expect(getUserCountry(state, '')).toEqual('United Kingdom')
    expect(getUserCountry(state, undefined)).toEqual('United Kingdom')
    expect(getUserCountry(state, null)).toEqual('United Kingdom')
  })

  describe('[selectStoredPaymentDetails selector]', () => {
    it('should extract creditCard from user', () => {
      snapshot(selectStoredPaymentDetails(state))
    })

    it('should return empty when state is empty', () => {
      expect(selectStoredPaymentDetails({})).toEqual({})
      expect(selectStoredPaymentDetails(null)).toEqual({})
      expect(selectStoredPaymentDetails(undefined)).toEqual({})
    })
  })

  describe('[getMyCheckoutDetailsEditingEnabled selector]', () => {
    it('false by default', () => {
      expect(getMyCheckoutDetailsEditingEnabled(state)).toBe(false)
    })
    it('true when changed', () => {
      const state = {
        account: {
          myCheckoutDetails: {
            editingEnabled: true,
          },
        },
      }
      expect(getMyCheckoutDetailsEditingEnabled(state)).toBe(true)
    })
    it('return false when state is empty', () => {
      expect(getMyCheckoutDetailsEditingEnabled({})).toEqual(false)
      expect(getMyCheckoutDetailsEditingEnabled(null)).toEqual(false)
      expect(getMyCheckoutDetailsEditingEnabled(undefined)).toEqual(false)
    })
  })

  describe('[getMyCheckoutDetailForm selector]', () => {
    const form = getMyCheckoutDetailForm(state)
    it('emtpy by default', () => {
      expect(form.fields.isDeliveryAndBillingAddressEqual).toEqual({
        value: '',
        isDirty: false,
        isTouched: false,
        isFocused: false,
      })
    })
    it('return null when state is empty', () => {
      expect(getMyCheckoutDetailForm({})).toEqual(null)
      expect(getMyCheckoutDetailForm(null)).toEqual(null)
      expect(getMyCheckoutDetailForm(undefined)).toEqual(null)
    })
  })

  describe('[getMCDPaymentMethodForm selector]', () => {
    it('get Card Details form', () => {
      const form = getMCDPaymentMethodForm(state)
      expect(form.fields).toBeDefined()
      expect(form.fields).toEqual(
        expect.objectContaining({
          paymentType: {
            value: 'VISA',
            isDirty: false,
            isTouched: false,
            isFocused: false,
          },
        })
      )
    })
  })

  describe(isSaveMyCheckoutDetailsDisabled.name, () => {
    it('return false when there are validation errors', () => {
      expect(isSaveMyCheckoutDetailsDisabled(myCheckoutDetailsErrors)).toEqual(
        false
      )
    })
    it('return false when there are no validation errors', () => {
      expect(
        isSaveMyCheckoutDetailsDisabled(myCheckoutDetailsNoErrors)
      ).toEqual(false)
    })
    it('return false when the only validation error is the CVV field', () => {
      expect(
        isSaveMyCheckoutDetailsDisabled(myCheckoutDetailsCVVError)
      ).toEqual(false)
    })
  })
  describe('selectDeliveryCountry selector', () => {
    it('should select the delivery country from checkout if present', () => {
      const state = {
        checkout: {
          orderSummary: {
            deliveryDetails: {
              address: {
                country: 'United Kingdom',
              },
            },
          },
        },
      }

      expect(selectDeliveryCountry(state)).toEqual('United Kingdom')
    })

    it('should select the delivery country from the checkout form if present', () => {
      const state = {
        forms: {
          checkout: {
            yourAddress: {
              fields: {
                country: {
                  value: 'United Kingdom',
                },
              },
            },
          },
        },
      }

      expect(selectDeliveryCountry(state)).toEqual('United Kingdom')
    })

    it('should select the delivery country from the user account if present and a checkout form is missing', () => {
      const state = {
        account: {
          user: {
            deliveryDetails: {
              address: {
                country: 'United Kingdom',
              },
            },
          },
        },
      }

      expect(selectDeliveryCountry(state)).toEqual('United Kingdom')
    })

    it('should select the delivery country from the config if all else fails', () => {
      const state = {
        config: {
          country: 'United Kingdom',
        },
      }

      expect(selectDeliveryCountry(state)).toEqual('United Kingdom')
    })
  })

  describe('selectBillingCountry selector', () => {
    it('should select the billing country from the checkout form if present', () => {
      const state = {
        forms: {
          checkout: {
            billingAddress: {
              fields: {
                country: {
                  value: 'United Kingdom',
                },
              },
            },
          },
        },
      }

      expect(selectBillingCountry(state)).toEqual('United Kingdom')
    })

    it('should select the billing country from the user account if present and a checkout form is missing', () => {
      const state = {
        account: {
          user: {
            billingDetails: {
              address: {
                country: 'United Kingdom',
              },
            },
          },
        },
      }

      expect(selectBillingCountry(state)).toEqual('United Kingdom')
    })

    it('should select the billing country from the config if all else fails', () => {
      const state = {
        config: {
          country: 'United Kingdom',
        },
      }

      expect(selectBillingCountry(state)).toEqual('United Kingdom')
    })
  })

  describe('getPaymentCardDetailsMCD', () => {
    it('should return payment card details for my checkout detail', () => {
      const paymentCardDetailsMCD = 'paymentCardDetailsMCD'
      const state = {
        forms: {
          account: {
            myCheckoutDetails: {
              paymentCardDetailsMCD,
            },
          },
        },
      }
      expect(getPaymentCardDetailsMCD(state)).toBe(paymentCardDetailsMCD)
    })

    it('should safely return undefined if state does not have forms', () => {
      expect(getPaymentCardDetailsMCD({})).toBeUndefined()
    })
  })

  describe('getOrderDetailsDeliveryPostCode', () => {
    it('Should return post code when passed state', () => {
      const state = {
        address4: 'SE7 R8E',
      }
      expect(getOrderDetailsDeliveryPostCode(state)).toEqual(state.address4)
    })
    it('Should safely return "" if state does not have address4', () => {
      expect(getOrderDetailsDeliveryPostCode({})).toEqual('')
    })
  })

  describe('getValidUKOrdersCountries', () => {
    it('Should return true if Country is United Kingdom, Jersey, Guernsey', () => {
      expect(getValidUKOrdersCountries({ country: 'United Kingdom' })).toBe(
        true
      )
      expect(getValidUKOrdersCountries({ country: 'Jersey' })).toBe(true)
      expect(getValidUKOrdersCountries({ country: 'Guernsey' })).toBe(true)
    })
    it('Should return false if Country is not United Kingdom, Jersey, Guernsey', () => {
      expect(getValidUKOrdersCountries({ country: 'India' })).toBe(false)
    })
  })

  describe('getValidOrderStatusReturn', () => {
    /**
     * M - order placed
     * C - order placed
     * W - order on hold
     * G - order shipment delayed
     * N - order payment pending approval
     * FLD - order payment failed
     * s - order partially shipped
     * L - order - low stock
     * r - item returned
     * i - parcel ready for collection
     */
    const orderStatusNotAllowed = [
      'M',
      'C',
      'W',
      'G',
      'N',
      'FLD',
      's',
      'L',
      'r',
      'i',
    ]

    /**
     * S - Order shipped
     * D - Order payment settled
     * c - Parcel collected by customer
     * */
    const orderStatusAllowed = ['S', 'D', 'c']

    it('Should return false if orderStatus does not comply ', () => {
      orderStatusNotAllowed.forEach((status) => {
        return expect(getValidOrderStatusReturn(status)).toBeFalsy()
      })
    })

    it('Should return true if orderStatus matches', () => {
      orderStatusAllowed.forEach((status) => {
        return expect(getValidOrderStatusReturn(status)).toBeTruthy()
      })
    })
  })

  describe('getCreditCardType', () => {
    it('Should return a credit card type', () => {
      expect(getCreditCardType(state)).toEqual('PYPAL')
    })

    it('Should return an empty string if state not available', () => {
      expect(getCreditCardType()).toEqual('')
    })
  })

  describe('isKlarnaDefaultPaymentType', () => {
    it('returns true if the default payment type is Klarna', () => {
      const state = {
        account: {
          user: {
            creditCard: {
              type: 'KLRNA',
            },
          },
        },
      }
      expect(isKlarnaDefaultPaymentType(state)).toEqual(true)
    })
    it('returns true if the default payment type is Klarna', () => {
      const state = {}
      expect(isKlarnaDefaultPaymentType(state)).toEqual(false)
    })
  })

  describe('isLoggedIn', () => {
    const stateNotLogged = {
      account: {
        user: {},
      },
    }
    it('return false if user not logged in', () => {
      expect(isLoggedIn(stateNotLogged)).toEqual(false)
    })
    it('return true if user logged in (exist: true)', () => {
      expect(isLoggedIn(state)).toEqual(true)
    })
  })

  describe(getExponeaMemberId.name, () => {
    const expId2 =
      '8ba233e437d8f896098ed097a0e2dc079c6908693cdec27af2b493865c0c9c76'

    const state = deepFreeze({
      account: {
        user: {
          expId2,
        },
      },
    })

    it('should return the ExponeaMemberId', () => {
      expect(getExponeaMemberId(state)).toBe(expId2)
    })

    it('should return an empty string if the ExponeaMemberId is not returned', () => {
      const newState = deepFreeze({
        ...state,
        account: {
          ...state.account,
          user: {},
        },
      })
      expect(getExponeaMemberId(newState)).toBe('')
    })
  })
})
