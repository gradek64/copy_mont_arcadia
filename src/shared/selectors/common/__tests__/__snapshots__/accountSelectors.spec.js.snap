// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@Account Selectors [get User Countries], extract billing country 1`] = `"United Kingdom"`;

exports[`@Account Selectors [get User Countries], extract delivery country 1`] = `"United Kingdom"`;

exports[`@Account Selectors [getUserAddress selector] extract user Address from billingDetails 1`] = `
Object {
  "address1": "7 Hannah Close",
  "address2": "Llanishen",
  "city": "CARDIFF",
  "country": "United Kingdom",
  "postcode": "hp3 9fs",
  "state": "",
}
`;

exports[`@Account Selectors [getUserAddress selector] extract user Address from deliveryDetails 1`] = `
Object {
  "address1": "7 Hannah Close",
  "address2": "Llanishen",
  "city": "CARDIFF",
  "country": "United Kingdom",
  "postcode": "se5 9hr",
  "state": "",
}
`;

exports[`@Account Selectors [getUserDetails selector] extract userDetails from user billingDetails 1`] = `
Object {
  "firstName": "new first name",
  "lastName": "Williams",
  "telephone": "07971134030",
  "title": "Mrs",
}
`;

exports[`@Account Selectors [getUserDetails selector] extract userDetails from user deliveryDetails 1`] = `
Object {
  "firstName": "new first name",
  "lastName": "Williams",
  "telephone": "07971134030",
  "title": "Mrs",
}
`;

exports[`@Account Selectors [isResetPasswordLinkValid selector return isResetPasswordLinkValid value 1`] = `false`;

exports[`@Account Selectors [selectStoredPaymentDetails selector] should extract creditCard from user 1`] = `
Object {
  "cardNumberHash": "tjOBl4zzS+toHslPB+u0Z0x24+iMu0Qb",
  "cardNumberStar": "************1152",
  "expiryMonth": "09",
  "expiryYear": "2018",
  "type": "PYPAL",
}
`;

exports[`@Account Selectors getDeliveryDetails returns the delivery details 1`] = `
Object {
  "address": Object {
    "address1": "7 Hannah Close",
    "address2": "Llanishen",
    "city": "CARDIFF",
    "country": "United Kingdom",
    "postcode": "se5 9hr",
    "state": "",
  },
  "addressDetailsId": 2245363,
  "nameAndPhone": Object {
    "firstName": "new first name",
    "lastName": "Williams",
    "telephone": "07971134030",
    "title": "Mrs",
  },
}
`;
